arch/RISCV/RISCVDisassembler.c

```diff
@@ -0,0 +1,432 @@
+//===-- RISCVDisassembler.cpp - Disassembler for RISCV --------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+/* Capstone Disassembly Engine */
+/* By Rodrigo Cortes Porto <porto703@gmail.com>, 2018 */
+
+#ifdef CAPSTONE_HAS_RISCV
+
+#include <stdio.h>		// DEBUG
+#include <stdlib.h>
+#include <string.h>
+
+#include "../../cs_priv.h"
+#include "../../utils.h"
+
+#include "../../MCInst.h"
+#include "../../MCInstrDesc.h"
+#include "../../MCFixedLenDisassembler.h"
+#include "../../MCRegisterInfo.h"
+#include "../../MCDisassembler.h"
+#include "../../MathExtras.h"
+#include "RISCVBaseInfo.h"
+#include "RISCVDisassembler.h"
+
+
+/* Need the feature infos define in 
+  RISCVGenSubtargetInfo.inc. */
+#define GET_SUBTARGETINFO_ENUM
+#include "RISCVGenSubtargetInfo.inc"
+
+/* When we specify the RISCV64 mode, It means It is RV64IMAFD.
+  Similar, RISCV32 means RV32IMAFD.
+*/
+static uint64_t getFeatureBits(int mode) 
+{
+	if (mode == CS_MODE_RISCV32)
+		// return b11110
+		return RISCV_FeatureStdExtM | RISCV_FeatureStdExtA |
+		       RISCV_FeatureStdExtF | RISCV_FeatureStdExtD ;
+	else
+		
+		// CS_MODE_RISCV64, return b11111
+		return RISCV_Feature64Bit   | RISCV_FeatureStdExtM | 
+		       RISCV_FeatureStdExtA | RISCV_FeatureStdExtF | 
+		       RISCV_FeatureStdExtD ;
+}
+
+#define GET_REGINFO_ENUM
+#define GET_REGINFO_MC_DESC
+#include "RISCVGenRegisterInfo.inc"
+#define GET_INSTRINFO_ENUM
+#include "RISCVGenInstrInfo.inc"
+
+///
```

Comment:
why do we have these lines /// in many places?

Reply:
all clear.

Reply:
But have the same x86 test error, I had merged the newest upstream/next already.

Reply:
you should fix that position of RISCV i mentioned above first, that could be the reason.

(remember to fix the RISCV position in all related arrays cs_arch_init, cs_arch_option & cs_arch_disallowed_mode_mask

Reply:
nice, now this is the remaining bug to fix, to have Travis pass on everything